ckanext/s3filestore/uploader.py

```diff
@@ -542,8 +542,18 @@ def __init__(self, resource):
             self.mimetype = resource.get('mimetype')
             if not self.mimetype:
                 try:
-                    # 512 bytes should be enough for a mimetype check
-                    self.mimetype = resource['mimetype'] = mime.from_buffer(self.upload_file.read(512))
+                    try:
+                        # Get type from file extension if we recognise it
+                        self.mimetype = mimetypes.guess_type(self.filename, strict=False)[0]
```

Comment:
what happens if someone changes the suffix, should we also not be checking that the file is 'valid' internally as well?

looking at 
https://github.com/ahupp/python-magic/blob/53aa709affef18cb2d1c5930f780dabfc5e4330c/README.md
# recommend using at least the first 2048 bytes, as less can produce incorrect identification

https://docs.python.org/3/library/mimetypes.html#mimetypes.guess_type

One of the good security models we have is that you can't upload an exe under a pdf or csv etc. Should we be removing this?

Reply:
That's in [ckanext-resource-type-validation](https://github.com/qld-gov-au/ckanext-resource-type-validation/), not here.

Reply:
ah, ok then, this is just setting the correct type. maybe added a comment in the code that this is for type setting, for file type validation, please ensure [ckanext-resource-type-validation](https://github.com/qld-gov-au/ckanext-resource-type-validation/) is added.